cdap-client/src/main/java/co/cask/cdap/client/MetaClient.java

```diff
@@ -53,7 +53,14 @@ public MetaClient(ClientConfig config) {
   }
 
   public void ping() throws IOException, UnauthorizedException {
-    restClient.execute(HttpMethod.GET, config.resolveURLNoVersion("ping"), config.getAccessToken());
+    HttpResponse response = restClient.execute(
+      HttpMethod.GET, config.resolveURLNoVersion("ping"), config.getAccessToken());
+    if (response.getResponseCode() != 200) {
```

Comment:
This approach won't work. Have we tested the CLI connecting to the UI? We get 200 OK for the ping endpoint hitting the UI host and port. We should check for the response body and make sure the response is  as expected. 

```
 curl -v http://localhost:9999/ping
* Hostname was NOT found in DNS cache
*   Trying 127.0.0.1...
* Connected to localhost (127.0.0.1) port 9999 (#0)
> GET /ping HTTP/1.1
> User-Agent: curl/7.37.1
> Host: localhost:9999
> Accept: */*
> 
< HTTP/1.1 200 OK
< X-Powered-By: Express
< Set-Cookie: bcookie=d0887e02-2e30-4614-8bf0-78f1b0bf30cc; Path=/; Expires=Wed, 23 Nov 2016 17:27:46 GMT
< Accept-Ranges: bytes
< Date: Tue, 24 Nov 2015 17:27:46 GMT
< Cache-Control: public, max-age=0
< Last-Modified: Wed, 21 Oct 2015 11:33:52 GMT
< ETag: W/"7a2-4057876526"
< Content-Type: text/html; charset=UTF-8
< Content-Length: 1954
< Vary: Accept-Encoding
< Connection: keep-alive
< 
* Connection #0 to host localhost left intact
 <!DOCTYPE html> <html lang="en" ng-app="cdap-ui" ng-strict-di> <head> <title ng-bind="$state.current | myTitleFilter">CDAP</title> <meta charset="utf-8"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <meta name="description" content="Cask Data Application Platform"> <meta name="viewport" content="width=device-width, initial-scale=1"> <base href="/"> <link rel="stylesheet" type="text/css" href="/assets/bundle/lib-f69cfe53.css"> <link rel="stylesheet" type="text/css" href="/assets/bundle/app-a64fed9a.css"> </head> <body ng-controller="BodyCtrl" ng-class="bodyClass"> <header class="navbar navbar-inverse navbar-fixed-top navbar-cdap" ng-if="!$state.includes('adapters.**') && !$state.includes('login')"> <div role="navigation" class="container" my-navbar></div> </header> <header class="navbar navbar-fixed-top navbar-hydrator" ng-if="$state.includes('adapters.**')"> <div role="navigation" class="container" my-navbar-hydrator></div> </header> <main class="container"> <div ui-view></div> </main> <div id="alerts"> <div class="container"></div> </div> <loading-icon></loading-icon> <footer> <div class="container"> <div class="row text-muted" ng-cloak> <div class="col-xs-12 col-md-6 text-uppercase"> <p> <img src="/assets/img/cask.png"> <span>Copyright &copy; 2015 Cask Data, Inc.</span> </p> </div> <div class="col-xs-12 col-md-6"> <ul class="list-inline"> <li> {{version}} </li> <li> <a href="http://cask.co/terms" target="_blank">Terms</a> </li> <li> <a href="http://cask.co/privacy" target="_blank">Privacy</a> </li> <li> <a href="http://cask.co/contact" target="_blank">Contact</a> </li> </ul> </div> </div> </div> </footer> <script type="text/javascript" src="/assets/bundle/lib-075b9157.js"></script> <script type="text/javascript" src="/config.js"></script> <script type="text/javascript" src="/assets/bundle/app-1a76dde9.js"></script> <script type="text/javascript" src="/assets/bundle/tpl-1ec32bf5.js"></script>  </body> </html> 
```